build-system/pr-check/esm-tests.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * This script builds and tests the AMP runtime in esm mode.
+ * This is run during the CI stage = test; job = esm tests.
+ */
+
+const colors = require('ansi-colors');
+const {
+  printChangeSummary,
+  startTimer,
+  stopTimer,
+  timedExecOrDie: timedExecOrDieBase,
+} = require('./utils');
+const {determineBuildTargets} = require('./build-targets');
+const {isTravisPullRequestBuild} = require('../common/travis');
+
+const FILENAME = 'esm-tests.js';
+const FILELOGPREFIX = colors.bold(colors.yellow(`${FILENAME}:`));
+const timedExecOrDie = (cmd, unusedFileName) =>
+  timedExecOrDieBase(cmd, FILENAME);
+
+function main() {
+  const startTime = startTimer(FILENAME, FILENAME);
+
+  if (!isTravisPullRequestBuild()) {
+    timedExecOrDie('gulp update-packages');
+    timedExecOrDie('gulp dist --fortesting --esm');
```

Comment:
No need to compile here. You can download what was built in the `ESM Dist` job from the `build` stage. This will shave off ~10 mins from the total build time.

https://github.com/ampproject/amphtml/blob/69134691e60ff8ba6aa72dbc4a0558da143db943/build-system/pr-check/utils.js#L338-L344

Recommended order of steps:

```js
downloadEsmDistOutput(FILENAME);
timedExecOrDie('gulp update-packages');
timedExecOrDie('gulp integration --nobuild --compiled --headless');
```